store/tikv/coprocessor.go

```diff
@@ -1361,17 +1374,22 @@ func (e *rateLimitAction) SetFallback(a memory.ActionOnExceed) {
 	e.fallbackAction = a
 }
 
-// broadcastIfNeeded will check whether the copWorkers is under suspended status.
-// If they are, `broadcastIfNeeded` would try to recover them if there are no more
-// copResponse remained in the channel.
+// broadcastIfNeeded will broadcast the condition to recover all suspended workers when exceeded is enabled
+// and one token have already been destroyed.
 func (e *rateLimitAction) broadcastIfNeeded(needed bool) {
+	if !needed {
+		return
+	}
 	e.conditionLock()
 	defer e.conditionUnlock()
-	if e.cond.exceeded && needed {
-		e.cond.exceeded = false
-		e.cond.Broadcast()
-		e.cond.once = sync.Once{}
```

Comment:
I remove initializing `sync.Once{}` in `broadcastIfNeeded` because I found a case which will cause deadlock if things happening in the following situation:

1. rateLimitAction execute Action
2. all workers being suspended
3. `broadcastIfNeeded` set exceed to false and broadcast the condition changing then initializing the once
4. rateLimitAction Action have been triggered again and successfully holding the mutex and execute Action again.
5. all workers holding mutex again and found the exceed is true, then become waiting and encounter deadlock.